qt.lua

```diff
@@ -575,6 +579,58 @@ function premake.extensions.qt.getQRCDependencies(fcfg)
 
 end
 
+--
+-- Checks if a file is a ts file.
+--
+-- @param filename
+--		The file name to check.
+-- @return
+--		true if the file needs to be run through the lrelease tool, false if not.
+--
+function premake.extensions.qt.isTS(filename)
+	return path.hasextension(filename, { ".ts" })
+end
+
+--
+-- Adds the custom build for a ts file.
+--
+-- @param fcfg
+--		The config for a single file.
+-- @param cfg
+--		The config of the project ?
+--
+function premake.extensions.qt.addTSCustomBuildRule(fcfg, cfg)
+	local qt = premake.extensions.qt
+
+	-- get the project's location (to make paths relative to it)
+	local projectloc = fcfg.project.location
+
+	-- create the output file name
+	local qtqmgenerateddir = fcfg.qtqmgenerateddir or cfg.qtqmgenerateddir
+	local output = path.join(qtqmgenerateddir or fcfg.directory, fcfg.basename .. ".qm")
+
+	-- create the moc command
+	local command = '"' .. fcfg.config.qtbinpath .. '/lrelease" "' .. fcfg.relpath .. '"'
+	command = command .. ' -qm "' .. path.getrelative(projectloc, output) .. '"'
+
+	-- now create the arguments
+	local arguments = ""
+
+	-- if we have custom commands, add them
+	local qttsargs = iif(#fcfg.qttsargs ~= 0, fcfg.qttsargs, cfg.qttsargs)
+	if #qttsargs ~= 0 then
+		table.foreachi(qttsargs, function (arg)
+			arguments = arguments .. ' "' .. arg .. '"'
+		end)
+		command = command .. arguments
+	end
+
+	-- add the custom build rule
+	fcfg.buildmessage = "lreleas'ing " .. fcfg.name
+	fcfg.buildcommands = iif(qtqmgenerateddir ~= nil, { "{MKDIR} " .. path.getrelative(projectloc, path.getdirectory(output)), command }, { command })
```

Comment:
I'm not very comfortable with running a `{MKDIR}` command for all `.ts` files. It feels a bit unnecessary: the creation of the output directory could be done during the workspace generation (command above) so that the build command could just be `{ command }` unconditionally. This would make the build command simpler, and would avoid unnecessary work during build time.
I'm open to discussion though, if you have any strong point to make for the `{MKDIR}` build command :)

Reply:
Removed currently.